lib/spotify/spotify.ex

```diff
@@ -0,0 +1,123 @@
+defmodule Spotify do
+  use Tesla
+
+  @base_url "https://api.spotify.com/v1"
+
+  plug(Tesla.Middleware.BaseUrl, @base_url)
+  plug(Tesla.Middleware.JSON)
+  # plug(Tesla.Middleware.Logger, debug: false, filter_headers: ["Authorization"])
+  plug(Spotify.OAuth.Middleware)
+
+  def find_track(query, opts \\ [limit: 5, remove_featuring: true]) do
+    sanitized = sanitize_search(query, opts[:remove_featuring])
+
+    case get!("/search", query: [q: sanitized, type: "track"] ++ Keyword.take(opts, [:limit])) do
+      %{status: 200, body: %{"tracks" => %{"items" => tracks}}} ->
+        track =
+          Enum.find(tracks, fn track ->
+            name = track["name"]
+            !String.contains?(name, ["Club Mix", "Karaoke"])
+          end) || List.first(tracks)
+
+        if track do
+          {:ok,
+           %{
+             search: sanitized,
+             original_query: query,
+             uri: track["uri"],
+             name: track["name"],
+             artists: Enum.map(track["artists"], fn a -> a["name"] end)
+           }}
+        else
+          {:err, %{search: sanitized, original_query: query, error: "Could not find any track"}}
+        end
+
+      err ->
+        {:err, %{search: sanitized, original_query: query, error: err}}
+    end
+  end
+
+  def create_playlist(user_id, name, info) do
+    playlist = playlist(name, info)
+
+    case post!("/users/#{user_id}/playlists", playlist) do
+      %{status: status, body: body} when status in [200, 201] -> body["id"]
+    end
+  end
+
+  defp playlist(playlist_name, info) do
+    %{
+      "name" => "Roule avec Driver - #{playlist_name}",
+      "description" => description(info)
+    }
+  end
+
+  def update_playlist_description(playlist_id, info) do
+    put!("/playlists/#{playlist_id}", %{"description" => description(info)})
+  end
+
+  defp description(%{title: title, link: link}) do
+    "Playlist generated from the Roule avec Driver youtube video \"#{title}\" [#{link}]. All credits to driver"
+  end
+
+  def update_playlist_tracks(playlist_id, tracks) do
+    put!("/playlists/#{playlist_id}/tracks", %{"uris" => tracks})
+  end
+
+  def _update_playlist_image(_playlist_id, _image_uri) do
+    # TODO
+    # put!("/playlists/#{playlist_id}/images", )
+  end
+
+  def get_playlist(playlist_id) do
+    get!("/playlists/#{playlist_id}")
+  end
+
+  def get_user() do
+    get!("/me")
+  end
+
+  # Remove feat, ft. Feat...
+  defp sanitize_search(query, remove_featuring) do
+    if remove_featuring do
+      String.replace(query, ~r/(\s+)[fF](ea)?t(.)?[^\\"]*/, " ")
+    else
+      String.replace(query, ~r/(\s+)[fF](ea)?t(.)?(\s+)/, "\\1")
+    end
+    |> String.replace(~r/(\s+)&?(\s+)/, " ")
+    |> String.replace("\"", "")
+    |> String.replace("(", "")
+    |> String.replace(")", "")
```

Comment:
[String.replace](https://hexdocs.pm/elixir/String.html#replace/4) accepts a list of strings